src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_validUser_ok() {
+        User user = new User();
+        user.setLogin("valid user");
+        user.setPassword("valid password");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+        assertEquals(user.getPassword(), registeredUser.getPassword());
+        assertEquals(user.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("validUser");
+        user.setPassword("123");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_underageUser_notOk() {
+        User user = new User();
+        user.setLogin("validUser");
+        user.setPassword("validPassword");
+        user.setAge(17);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_existingUserLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("validUser");
+        user1.setPassword("validPassword");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("validUser");
+        user2.setPassword("anotherPassword");
+        user2.setAge(22);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user2);
+        });
+    }
```

Comment:
The test `register_existingUserLogin_notOk` assumes that `register` method works correctly for adding users. Use a low-level method to add `user1` to the storage instead of the `register` method.